packages/report-server/src/__tests__/reportBuilder/transform/insertRows.test.ts

```diff
@@ -163,29 +186,35 @@ describe('insert', () => {
   it('can sum all values into a new totals row', () => {
     const transform = buildTransform([
       {
-        transform: 'insert',
-        where: 'eq($index, length($table))',
-        "'Total'": 'sum($all.value)',
+        transform: 'insertRows',
+        where: 'eq(@index, length(@table))',
+        columns: {
+          Total: 'sum(@all.value)',
+        },
       },
     ]);
     expect(transform(MULTIPLE_ANALYTICS)).toEqual([
       { period: '20200101', organisationUnit: 'TO', dataElement: 'BCD1', value: 4 },
       { period: '20200102', organisationUnit: 'TO', dataElement: 'BCD1', value: 2 },
       { period: '20200103', organisationUnit: 'TO', dataElement: 'BCD1', value: 5 },
-      { Total: 11 }
+      { Total: 11 },
     ]);
   });
 
   it('compare adjacent rows to insert between', () => {
     const transform = buildTransform([
       {
-        transform: 'insert',
-        where: 'not(eq($row.organisationUnit, $next.organisationUnit))',
-        "'Total_BCD1'": "sum($where(f($otherRow) = equalText($otherRow.organisationUnit, $row.organisationUnit)).BCD1)",
-        "'Total_BCD2'": "sum($where(f($otherRow) = equalText($otherRow.organisationUnit, $row.organisationUnit)).BCD2)",
+        transform: 'insertRows',
+        where: 'not(eq($organisationUnit, @next.organisationUnit))',
+        columns: {
+          Total_BCD1:
+            'sum(@where(f(@otherRow) = equalText(@otherRow.organisationUnit, $organisationUnit)).BCD1)',
```

Comment:
I have a few problems with the `where`...  Main one is that we need to use the function syntax with it! However I think I found a way to avoid that if we create it as a `rawFunction` (https://mathjs.org/docs/expressions/customization.html#custom-argument-parsing)

I agree that `@where` doesn't fit with the others here... Possible suggestion is that `where` is the base function, and the other lookups are properties of that function, eg.
```
cost whereFunction = () => { ... do the where logic };
const lookup = whereFunction;
lookup.current = row;
lookup.previous = lastRow;
...etc

parser.set('@lookup', lookup);

So then in a report it would go:

{
  transform: 'updateColumns',
  insert: {
    rowValue: '@lookup.current.value',
    lastRowValue: '@lookup.previous.value',
    sumMatchingRowsValue: 'sum(@lookup(f(@otherRow) = equalText(@otherRow.organisationUnit, $organisationUnit)).value'
  }
}
```

Reply:
Interesting idea...I think I prefer the existing, shorter syntax for those simple lookups.

I'm actually fine with `where` using the function syntax, _if_ we don't also use the `@`. Can we just make `where` a standard function?

Reply:
(btw not sold on the term `lookup`, could be anything, or even just `@`?)

Reply:
Aaaah now that's interesting... `@` _is_ `where`, and there are some shortcuts into `@previous` etc. but also `@(...)`

I'm still wondering if there's anything wrong with just `where(...)` with no prefix though?

Reply:
I think I might just have a hangup on _lookups_ being this special thing to reference other data in the table, and thus needing to have a special syntax. I think that's probably not a good justification for keeping where being special haha! 

Would having both bother you? I can see the argument against it since it's usually not wise having two ways to do the one thing?

Reply:
I'd prefer to go with the special syntax than both I think, agree that we're better to be opinionated than allow multiple options here

Reply:
Definitely happy to chat about this one with others and change it later on! Kinda just keen to get these changes out at this point as I feel we can discuss semantics til the cows come home! 🐮 

Also worth noting that where is a pretty complex operation and I hope most people aren't using it! If it's in heavy use I feel it's symptomatic of report-server needing more specific built in functions!

Reply:
Okay took one look at the 'special syntax' for where and in it's current form it's pretty unreadable (**LOTS** of symbols 😝 ) am going with `where` !